app/script/conversation/ConversationRepository.js

```diff
@@ -156,6 +156,8 @@ z.conversation.ConversationRepository = class ConversationRepository {
     this._init_subscriptions();
 
     this.stateHandler = new z.conversation.ConversationStateHandler(this.conversation_service, this);
+    this.timedMessageHandler = new z.conversation.ConversationTimedHandler(this.conversation_service, this);
```

Comment:
This circular dependency (the fact that the `ConversationTimedHandler` depends on the `ConversationRepository` and vice versa) trigger a huge warning. 

It means the concerns are not clearly separated and that those two services should be siblings, not parent>child.

Since I am working on solving this issue right now, lets sit together and talk about this, WDYT?

Reply:
Yes, absolutely, the handlers are all siblings. My problem is with the fact that we inject the `repository` in the `handler` (meaning that there is a bidirectional communication between the two).

We came up with a nice solution to that yesterday with @AndyLnd that completely get rid of the circular dependency.